JoinColony/colonyDapp
# Refactor Add Existing Safe flow

## Description

This PR refactors the add existing safe form. 

There's one outstanding todo which is what to do with chains that don't have bridges. I will add the change to this PR or to a follow-up PR when I have more info but don't envision it being a significant change so want to get the review process rolling in the meantime. 

## To test
1. You need a safe deployed.
2. You need the zodiac bridge module connected to that safe. ([Follow this tutorial](https://gnosis.github.io/zodiac/docs/tutorial-module-bridge/get-started))

The following info can be used if you don't have your own (Rinkeby): 
Safe 1: `0x93ebb9b7b0a3d42d3AF76930EF569F196e3821C1`, Module 1: `0x25603925a75D39AFFb22e810aC4a2A92BCfEe6B4`
Safe 2: `0xF8F46F65b7365b97691C6794d0ff235B05BFAf78`, Module 2: `0xcae11900da49EB16a14D20f31237E04542F8Da35`

You should be able to change the inputs and it should always display the correct status / error, and you should be able to move forwards and backwards through the form without state changing or any other funny business. Submitting the form should take you to the actions page, where you should see all the info including the `moduleAddress`.

**New stuff** ✨

* `CheckSafe, ConnectSafe, ConfirmSafe`
* `DetailsWidgetAddSafe` (extracting logic to own component)
* `useClipboardCopy` (extracting logic to own hook)

**Changes** 🏗

* ActionsPage now includes `moduleAddress` in Details Widget
* Add onblur to core `Input` component (did not work previously. Used to set loading state in `CheckSafe` when field is not touched (i.e. first time user enters an address))
* Moved address validations into the yup schema in `AddExistingSafeDialog` (where they belong) instead of having them in effect hooks in the form. 

## Screenshots
![1](https://user-images.githubusercontent.com/64402732/191061622-192e58a6-501f-4ac8-847b-e507d2e50f84.png)
![2](https://user-images.githubusercontent.com/64402732/191061633-542c97d6-f447-44a7-abf0-5b2f1234dedf.png)
![step3](https://user-images.githubusercontent.com/64402732/191061642-ea5e60ff-de83-481f-a077-46a373f0cd4c.png)
![deets](https://user-images.githubusercontent.com/64402732/191066493-75f165fb-5c17-4937-b5e8-6bdc852b00df.png)

Resolves #3849
